File: ecommerce-front/src/app/interfaces/ShoppingCart.ts
import { OrderItem as OrderItem } from './OrderItem';
import { UserAggregate } from './UserAggregate';

export class ShoppingCart {
  ShoppingCartId?: number;
  UserAggregateId!: number;
  UserAggregate!: UserAggregate;
  OrderItems!: OrderItem[];
}

File: ByteStore.Domain/Aggregates/ShoppingCart.cs
using ByteStore.Domain.Entities;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace ByteStore.Domain.Aggregates;

public class ShoppingCart
{
    [Key] [Column("Id")] 
    public int ShoppingCartId { get; set; }
    public int UserAggregateId { get; set; }
    public UserAggregate UserAggregate { get; set; }
    public byte[]? OrderItems { get; set; }
}

